### PR TITLE
Implement dynamodb streams connector

### LINKED ISSUE #5801: Implement the dynamodb streams connector
**Description:**
subtask of https://github.com/ballerina-platform/ballerina-library/issues/4830

### PR DESCRIPTION
**Pull request by Bhashinee**
## Purpose
Implement the DynamoDB Streams connector.
Fixes : https://github.com/ballerina-platform/ballerina-library/issues/5801

### CODE REVIEW

File: .github/workflows/build-with-bal-test-native.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: GraalVM Check
```
Comment (daneshk) [inline .github/workflows/build-with-bal-test-native.yml]: Do we need both native and graalvm workflows?

Comment (Bhashinee) [reply .github/workflows/build-with-bal-test-native.yml]: We don't need it. Removed with [2cc8a76](https://github.com/ballerina-platform/module-ballerinax-aws.dynamodbstreams/pull/2/commits/2cc8a76c20c0d3cee16f82892cc2b39ee6eea76b)

File: ballerina/client.bal [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerinax/'client.config;
+
+# The Ballerina AWS DynamoDB Streams connector provides the capability to access AWS DynamoDb Streams related operations.
```
Comment (daneshk) [inline ballerina/client.bal]:
```suggestion
# The Ballerina AWS DynamoDB Streams connector provides the capability to access AWS DynamoDB Streams related operations.
```

File: ballerina/client.bal [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    private final string region;
+    private final string awsHost;
+    private final string uri = SLASH;
+
+    # Initializes the connector. During initialization you have to pass access key id, secret access key, and region.
+    # Create an AWS account and obtain tokens following
+    # [this guide](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html).
+    #
+    # + awsDynamoDBConfig - Configuration required to initialize the client
+    # + httpConfig - HTTP configuration
```
Comment (daneshk) [inline ballerina/client.bal]:
```suggestion
    # + config - Configuration required to initialize the client
```

File: ballerina/constatnts.bal [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+const string HEADER_CONTENT_TYPE = "Content-Type";
+const string HEADER_X_AMZ_CONTENT_SHA256 = "X-Amz-Content-Sha256";
+const string HEADER_X_AMZ_DATE = "X-Amz-Date";
+const string HEADER_X_AMZ_TARGET = "X-Amz-Target";
+const string HEADER_HOST = "Host";
+const string HEADER_AUTHORIZATION = "Authorization";
+
+const string GENERATE_SIGNED_REQUEST_HEADERS_FAILED_MSG = "Error occurred while generating signed request headers.";
+
+public enum AttributeType {
```
Comment (daneshk) [inline ballerina/constatnts.bal]: Shall we add API docs for public enums. This will appear in API Docs.

File: ballerina/stream_implementor.bal [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        self.secretAccessKey = secretKey;
+        self.region = region;
+        self.awsHost = AWS_STREAMS_SERVICE + DOT + self.region + DOT + AWS_HOST;
+        self.lastEvaluatedStreamArn = null;
+        self.listStreamInput = streamInput;
+        self.currentEntries = check self.fetchStreams();
+    }
+
+    public isolated function next() returns record {| Stream value; |}|error? {
+        if (self.index < self.currentEntries.length()) {
```
Comment (daneshk) [inline ballerina/stream_implementor.bal]:
```suggestion
        if self.index < self.currentEntries.length() {
```

File: examples/order-management/client.bal [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import ballerina/io;
```
Comment (daneshk) [inline examples/order-management/client.bal:17]: shall we add license headers?

File: ballerina/constatnts.bal [hunk 7]
```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```
Comment (daneshk) [inline ballerina/constatnts.bal]: typo in the filename `constants.bal`

File: ballerina/stream_implementor.bal (hunk 5 above)
Comment (daneshk) [reply ballerina/stream_implementor.bal]: please check other places

File: ballerina/constatnts.bal (hunk 4 above)
Comment (Bhashinee) [reply ballerina/constatnts.bal]: Added docs for enum types.

File: ballerina/stream_implementor.bal (hunk 5 above)
Comment (Bhashinee) [reply ballerina/stream_implementor.bal]: Fixed in [9a61dc7](https://github.com/ballerina-platform/module-ballerinax-aws.dynamodbstreams/pull/2/commits/9a61dc79f52588cd3098b3a576772e734b616e93)

File: examples/order-management/client.bal (hunk 6 above)
Comment (Bhashinee) [reply examples/order-management/client.bal:17]: Added in [9a61dc7](https://github.com/ballerina-platform/module-ballerinax-aws.dynamodbstreams/pull/2/commits/9a61dc79f52588cd3098b3a576772e734b616e93)

File: ballerina/constatnts.bal (hunk 7 above)
Comment (Bhashinee) [reply ballerina/constatnts.bal]: Renamed the file.

**Review by daneshk — APPROVED** [review]

**Pull request merged**